sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -0,0 +1,615 @@
+        public void PublishPartitionEventsToChannelComplete(string entityName,
+                                                            string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(19, entityName ?? string.Empty, partitionId ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while reading events from an Entity partition.
+        /// </summary>
+        ///
+        /// <param name="entityName">The name of the Entity that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(20, Level = EventLevel.Error, Message = "An exception occurred while publishing events to a background channel for Entity: {0} (Partition Id: '{1}'). Error Message: '{2}'")]
+        public void PublishPartitionEventsToChannelError(string entityName,
+                                                         string partitionId,
+                                                         string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(20, entityName ?? string.Empty, partitionId ?? string.Empty, errorMessage ?? string.Empty);
+            }
+        }
+        /// <summary>
+        ///   Indicates that reading events from an Entity partition has started.
+        /// </summary>
+        ///
+        /// <param name="entityName">The name of the Entity that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        ///
+        [Event(21, Level = EventLevel.Informational, Message = "Beginning to read events for Entity: {0} (Partition Id: '{1}').")]
+        public void ReadEventsFromPartitionStart(string entityName,
+                                                 string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(21, entityName ?? string.Empty, partitionId ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that reading events from an Entity partition has completed.
+        /// </summary>
+        ///
+        /// <param name="entityName">The name of the Entity that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        ///
+        [Event(22, Level = EventLevel.Informational, Message = "Completed reading events for Entity: {0} (Partition Id: '{1}').")]
+        public void ReadEventsFromPartitionComplete(string entityName,
+                                                    string partitionId)
```

Comment:
I believe these methods are the copy/paste remnants from Event Hubs land? we need to update these as per service bus methods.

Reply:
Yeah we will need to go through and update the event sources.  I will create an issue for this.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/9875